sdk/storage/Azure.Storage.Queues/src/QueueClientOptions.cs

```diff
@@ -113,6 +113,23 @@ public QueueClientOptions(ServiceVersion version = LatestVersion)
         /// </summary>
         public QueueMessageEncoding MessageEncoding { get; set; } = QueueMessageEncoding.None;
 
+        /// <summary>
+        /// Optional. Performs the tasks needed when an invalid message is received or peaked from the queue.
+        ///
+        /// <para>Invalid message can be received or peaked when <see cref="QueueClient"/> is expecting certain <see cref="QueueMessageEncoding"/>
+        /// but there's another producer that is not encoding messages in expected way. I.e. the queue contains messages with different encoding.</para>
+        ///
+        /// <para><see cref="InvalidMessageEventArgs"/> contains <see cref="QueueClient"/> that has received invalid message as well as the message
+        /// which can be either <see cref="QueueMessage"/> or <see cref="PeekedMessage"/> with raw body, i.e. no decoding will be attempted so that
+        /// body can be inspected as has been received from the queue.</para>
+        ///
+        /// <para>The <see cref="QueueClient"/> won't attempt to remove invalid message from the queue. Therefore such handling should be included into
+        /// the event handler itself.</para>
+        /// </summary>
+        public event SyncAsyncEventHandler<InvalidMessageEventArgs> OnInvalidMessage;
```

Comment:
This isn't a valid event name in .NET.  Please see https://learning.oreilly.com/library/view/framework-design-guidelines/9780135896457/ch03.xhtml#sec3_6_3.  What about `MessageDecodingFailed`?  We might need to rename the event too.